app/src/main/java/com/kamron/pogoiv/OcrHelper.java

```diff
@@ -144,6 +144,35 @@ private double getPokemonLevelFromImg(Bitmap pokemonImage, int trainerLevel) {
     }
 
     /**
+     * Get the evolution cost for a pokemon, example, weedle : 12
+     * If there was no detected upgrade cost, returns -1
+     *
+     * @param pokemonImage The image of the full pokemon screen
+     * @return the evolution cost, or -1 on scan failure
+     */
+    public int getPokemonEvolutionCostFromImg(Bitmap pokemonImage) {
+        Bitmap evolutionCostImage =
+                Bitmap.createBitmap(pokemonImage, (int) (widthPixels * 0.625), (int) (heightPixels * 0.86),
+                        (int) (widthPixels * 0.2), (int) (heightPixels * 0.05));
+        String hash = "candyCost" + hashBitmap(evolutionCostImage);
+        String stringCacheEvoCandyCost = ocrCache.get(hash);
+        int result = -1;
+        if (stringCacheEvoCandyCost == null) { //if no cached result
+            //the dark color used for text in pogo is 76,112,114
+            evolutionCostImage = replaceColors(evolutionCostImage, 76, 112, 114, Color.WHITE, 40, false);
+            tesseract.setImage(evolutionCostImage);
+            stringCacheEvoCandyCost = tesseract.getUTF8Text().replace("S", "5").replace("s", "5");
+            try {
+                result = Integer.parseInt(stringCacheEvoCandyCost);
+            } catch (NumberFormatException e) {
+                stringCacheEvoCandyCost = "-1";
+            }
+            ocrCache.put(hash, stringCacheEvoCandyCost);
+        }
```

Comment:
Wait, maybe it won't. In any case, you fixed it later for this one. I'll wait and see what other issues are reported. I should be able to fix most. But I'll do it in batches of 24hrs of reports.